src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,145 @@
-
+        putVal(hash(key), key, value);
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> node = getNode(key);
+        if (node == null) {
+            return null;
+        } else {
+            return node.value;
+        }
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void putVal(int hash, K key, V value) {
+        Node<K, V> newNode = new Node<>(hash, key, value, null);
+        int index = getIndexFromHash(hash);
+
+        if (table[index] == null) {
+            table[index] = newNode;
+        } else {
+            Node<K, V> node = table[index];
+            do {
+                if (node.hash == hash(key) && Objects.equals(key, node.key)) {
+                    node.value = value;
+                    return;
+                }
+                if (node.next != null) {
+                    node = node.next;
+                }
+            } while (node.next != null);
+            node.next = newNode;
+        }
+        if (++size > threshold) {
+            resize();
+        }
+    }
+
+    private int getIndexFromHash(int hash) {
+        return Math.abs(hash % table.length);
+    }
+
+    private void resize() {
+        Node<K, V>[] oldTable = table;
+        int oldTableLength = table.length;
+        int newLength = 0;
+
+        if (oldTableLength >= MAXIMUM_CAPACITY) {
+            threshold = Integer.MAX_VALUE;
+        } else {
+            newLength = oldTableLength * GROW_CAPACITY_FACTOR;
+            threshold = (int) (newLength * DEFAULT_LOAD_FACTOR);
+        }
+
+        table = new Node[newLength];
+        relinkNodes(oldTable, oldTableLength);
+    }
+
+    private void relinkNodes(Node<K, V>[] oldTable, int oldTableLength) {
+        for (int j = 0; j < oldTableLength; j++) {
+            Node<K, V> node = oldTable[j];
+            if (node != null) {
+                oldTable[j] = null;
+                int index = getIndexFromHash(node.hash);
+
+                if (node.next == null) {
+                    table[index] = node;
+                } else {
+                    while (node != null) {
+                        index = getIndexFromHash(node.hash);
+                        Node<K, V> nodeInsertionPos = table[index];
+                        if (nodeInsertionPos == null) {
+                            table[index] = node;
+                        } else {
+                            while (nodeInsertionPos.next != null) {
+                                nodeInsertionPos = nodeInsertionPos.next;
+                            }
+                            nodeInsertionPos.next = node;
+                        }
+                        nodeInsertionPos = node;
+                        node = node.next;
+                        nodeInsertionPos.next = null;
+                    }
+                }
+            }
+        }
+    }
+
+    private Node<K, V> getNode(K key) {
```

Comment:
The method `getNode` should be after all public methods according to the checklist. Please relocate it accordingly.